File: ts/datastructure/otterop/datastructure/LinkedListNode.ts
import { LinkedList } from './LinkedList';

export class LinkedListNode<T> {

    #list : LinkedList<T>;

    #prev : LinkedListNode<T>;

    #next : LinkedListNode<T>;

    #value : T;

    public constructor(value : T) {
        this.#prev = null;
        this.#next = null;
        this.#value = value;
    }

    public list() : LinkedList<T> {
        return this.#list;
    }

    /** @internal */
    setList(list : LinkedList<T>) : void {
        this.#list = list;
    }

    public prev() : LinkedListNode<T> {
        return this.#prev;
    }

    /** @internal */
    setPrev(node : LinkedListNode<T>) : void {
        this.#prev = node;
    }

    public next() : LinkedListNode<T> {
        return this.#next;
    }

    /** @internal */
    setNext(node : LinkedListNode<T>) : void {
        this.#next = node;
    }

    public value() : T {
        return this.#value;
    }
}

File: dotnet/src/Datastructure/Otterop/Datastructure/LinkedListNode.cs
namespace Otterop.Datastructure
{
    public class LinkedListNode<T>
    {
        private LinkedList<T> list;
        private LinkedListNode<T> prev;
        private LinkedListNode<T> next;
        private T value;
        public LinkedListNode(T value)
        {
            this.prev = null;
            this.next = null;
            this.value = value;
        }

        public LinkedList<T> List()
        {
            return this.list;
        }

        internal void SetList(LinkedList<T> list)
        {
            this.list = list;
        }

        public LinkedListNode<T> Prev()
        {
            return this.prev;
        }

        internal void SetPrev(LinkedListNode<T> node)
        {
            this.prev = node;
        }

        public LinkedListNode<T> Next()
        {
            return this.next;
        }

        internal void SetNext(LinkedListNode<T> node)
        {
            this.next = node;
        }

        public T Value()
        {
            return this.value;
        }

    }

}
